### PR TITLE
server rendering

### PR DESCRIPTION
**Pull request by yoshuawuyts**
Wrote a bunch about server rendering. Figured we should just cover the basics first, as we don't quite have a story for async rendering. Hope this works out alright!

### CODE REVIEW

File: content/reference/server-rendering.md [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+And then make sure the page becomes interactive quickly. And once that's done,
+you want it to all keep feeling snappy. And then some.
+
+Server rendering is a technique to improve time to first render in applications
+that rely on JavaScript to update the DOM. Instead of waiting for JavaScript to
+render in the client, it figures out what the DOM should look like on the
+server. The server then sends the right HTML to the client, so the first render
+happens instantly.
+
+Choo was built with both Node and the Browser in mind. One of its main feature
```
Comment (ghost) [inline content/reference/server-rendering.md]: `features`

File: content/reference/server-rendering.md [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+2. Create a new route, it returns some HTML.
+3. Here is where the rendering happens. We call our application with
+   `.toString(route)`, and it returns a string. The resulting string will be
+   `'<body>Hello humans</body>'`.
+
+## Adding in Data
+There hasn't been too much going on so far. Instead of rendering to the DOM,
+we've been creating strings instead. However, what's often the case with server
+rendering is that you need some data passed in for your initial rendering.
+Luckily the `.toString()` method accepts a seonc argument: `state`. Let's see
```
Comment (ghost) [inline content/reference/server-rendering.md]: `second`

File: content/reference/server-rendering.md [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+2. If the current file is required by some other file, we export it instead.
+
+### Modern
+The `import` keyword is what's going to be used to load JavaScript modules in
+the near future. It already works in Node, but isn't quite prevalent yet.
+Because the details of `import` are different from `require`, using it requires
+a slightly different approach.
+
+```js
+var choo = require('choo')
```
Comment (ghost) [inline content/reference/server-rendering.md]: isn't it supposed to use `import` here then?

**Review by ghost — DISMISSED** [review]
very good!

File: content/reference/server-rendering.md (hunk 3 above)
Comment (yoshuawuyts) [reply content/reference/server-rendering.md]: lol yup

**Review by ghost — APPROVED** [review]

**Pull request merged**